src/Nethermind/Nethermind.TxPool/TxBroadcaster.cs

```diff
@@ -97,31 +97,31 @@ internal class TxBroadcaster : IDisposable
         // only for testing reasons
         internal Transaction[] GetSnapshot() => _persistentTxs.GetSnapshot();
 
-        public void Broadcast(Transaction tx, bool isPersistent)
+        public bool Broadcast(Transaction tx, bool isPersistent)
         {
             if (isPersistent)
             {
-                StartBroadcast(tx);
-            }
-            else
-            {
-                BroadcastOnce(tx);
+                return StartBroadcast(tx);
             }
+
+            BroadcastOnce(tx);
+
+            return true;
         }
 
-        private void StartBroadcast(Transaction tx)
+        private bool StartBroadcast(Transaction tx)
         {
             // broadcast local tx only if MaxFeePerGas is not lower than configurable percent of current base fee
             // (70% by default). Otherwise only add to persistent txs and broadcast when tx will be ready for inclusion
-            if (tx.MaxFeePerGas >= _baseFeeThreshold || tx.IsFree())
+            if (tx is not null
+                && _persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx)
+                && (tx.MaxFeePerGas >= _baseFeeThreshold || tx.IsFree()))
```

Comment:
Ahh but order is wrong, we are firstly trying to insert and then checking fee, so it is possible that we will add tx to persistent collection, but return false. I made this mistake in proposed code in previous review
```suggestion
            if (tx is not null
                && (tx.MaxFeePerGas >= _baseFeeThreshold || tx.IsFree())
                && _persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx))
```

Reply:
And there is one more case, thare is a possibility that we will insert tx and evict it just after, so we need logic similar to the one from TxPool. Modify
`_persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx)`
to be
`_persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx, out var removed)`
and then check if `tx.Hash` is the same as `removed.Hash`. If hashes are the same it means that we didn't actually added tx to persistent collection

It is because we have some capacity, but under the hood we are actually firstly adding new tx (and having capacity max + 1) and then removing the worst one. So if we add and remove the same tx, it means that it wasn't actually added :D

Reply:
and then tests should work as expected, this is the reason why tests are failing